lighthouse-core/gather/driver/navigation.js

```diff
@@ -0,0 +1,109 @@
+const {waitForFullyLoaded, waitForFrameNavigated} = require('./wait-for-condition.js');
+const constants = require('../../config/constants.js');
+
+// Controls how long to wait after FCP before continuing
+const DEFAULT_PAUSE_AFTER_FCP = 0;
+// Controls how long to wait after onLoad before continuing
+const DEFAULT_PAUSE_AFTER_LOAD = 0;
+// Controls how long to wait between network requests before determining the network is quiet
+const DEFAULT_NETWORK_QUIET_THRESHOLD = 5000;
+// Controls how long to wait between longtasks before determining the CPU is idle, off by default
+const DEFAULT_CPU_QUIET_THRESHOLD = 0;
+
+/** @typedef {{waitUntil: Array<'fcp'|'load'|'navigated'>} & LH.Config.SharedPassNavigationJson & Partial<Pick<LH.Config.Settings, 'maxWaitForFcp'|'maxWaitForLoad'>>} NavigationOptions */
+
+/** @param {NavigationOptions} options */
+function resolveWaitForFullyLoadedOptions(options) {
+  /* eslint-disable max-len */
+  let {pauseAfterFcpMs, pauseAfterLoadMs, networkQuietThresholdMs, cpuQuietThresholdMs} = options;
+  let maxWaitMs = options.maxWaitForLoad;
+  let maxFCPMs = options.maxWaitForFcp;
+
+  if (typeof pauseAfterFcpMs !== 'number') pauseAfterFcpMs = DEFAULT_PAUSE_AFTER_FCP;
+  if (typeof pauseAfterLoadMs !== 'number') pauseAfterLoadMs = DEFAULT_PAUSE_AFTER_LOAD;
+  if (typeof networkQuietThresholdMs !== 'number') {
+    networkQuietThresholdMs = DEFAULT_NETWORK_QUIET_THRESHOLD;
+  }
+  if (typeof cpuQuietThresholdMs !== 'number') cpuQuietThresholdMs = DEFAULT_CPU_QUIET_THRESHOLD;
+  if (typeof maxWaitMs !== 'number') maxWaitMs = constants.defaultSettings.maxWaitForLoad;
+  if (typeof maxFCPMs !== 'number') maxFCPMs = constants.defaultSettings.maxWaitForFcp;
+  /* eslint-enable max-len */
+
+  if (!options.waitUntil.includes('fcp')) maxFCPMs = undefined;
+
+  return {
+    pauseAfterFcpMs,
+    pauseAfterLoadMs,
+    networkQuietThresholdMs,
+    cpuQuietThresholdMs,
+    maxWaitForLoadedMs: maxWaitMs,
+    maxWaitForFcpMs: maxFCPMs,
+  };
+}
+
+/**
+ * Navigates to the given URL, assuming that the page is not already on this URL.
+ * Resolves on the url of the loaded page, taking into account any redirects.
+ * Typical use of this method involves navigating to a neutral page such as `about:blank` in between
+ * navigations.
+ *
+ * @param {LH.Gatherer.FRTransitionalDriver} driver
+ * @param {string} url
+ * @param {NavigationOptions} options
+ * @return {Promise<{finalUrl: string, timedOut: boolean}>}
+ */
+async function gotoURL(driver, url, options) {
+  const session = driver.defaultSession;
+  const networkMonitor = new NetworkMonitor(driver.defaultSession);
+
+  // Enable the events and network monitor needed to track navigation progress.
+  await networkMonitor.enable();
+  await session.sendCommand('Page.enable');
+  await session.sendCommand('Page.setLifecycleEventsEnabled', {enabled: true});
+
+  // No timeout needed for Page.navigate. See https://github.com/GoogleChrome/lighthouse/pull/6413.
```

Comment:
super nit
```suggestion
  // No timeout needed for Page.navigate. See https://github.com/GoogleChrome/lighthouse/pull/6413
```

That damn period messes up the link when I click in GitHub.

Reply:
see @connorjclark this whole period business was a mistake 😉